metacat-metadata-mysql/src/main/java/com/netflix/metacat/metadata/mysql/MysqlUserMetadataService.java

```diff
@@ -191,12 +190,25 @@ private void _deleteDefinitionMetadatas(
         @Nullable final List<QualifiedName> names
     ) {
         if (names != null && !names.isEmpty()) {
-            final List<String> paramVariables = names.stream().map(s -> "?").collect(Collectors.toList());
-            final SqlParameterValue[] aNames = names.stream().map(n -> new SqlParameterValue(Types.VARCHAR, n))
+            final SqlParameterValue[] aNames = names.stream().filter(name -> !name.isPartitionDefinition())
+                .map(n -> new SqlParameterValue(Types.VARCHAR, n))
                 .toArray(SqlParameterValue[]::new);
-            jdbcTemplate.update(
-                String.format(SQL.DELETE_DEFINITION_METADATA, Joiner.on(",").skipNulls().join(paramVariables)),
-                (Object[]) aNames);
+            final SqlParameterValue[] aPartitionNames = names.stream().filter(QualifiedName::isPartitionDefinition)
+                .map(n -> new SqlParameterValue(Types.VARCHAR, n))
+                .toArray(SqlParameterValue[]::new);
+            if (aNames.length > 0) {
+                final List<String> paramVariables = Arrays.stream(aNames).map(s -> "?").collect(Collectors.toList());
+                jdbcTemplate.update(
+                    String.format(SQL.DELETE_DEFINITION_METADATA, Joiner.on(",").skipNulls().join(paramVariables)),
```

Comment:
Is this table definition metadata only or including database metadata? If just table only, we should name it explicitly as table definition metadata.

Reply:
Mentioned in the description. Will add it in the comments in the class.